firefly/client.py

```diff
@@ -75,6 +75,8 @@ def is_file(self, value):
     def handle_response(self, response):
         if response.status_code == 200:
             return self.decode_response(response)
+        elif response.status_code == 400:
+            raise ValueError(response.json()["error"])
```

Comment:
What if response is not JSON encoded? what if there is no error field in the data?

Reply:
The error generation ensures that.

`firefly.app` Line 104-108
```
        try:
            kwargs = self.get_inputs(request)
        except ValueError as err:
            logger.warn("Function %s failed with ValueError: %s.", self.name, err)
            return self.make_response({"error": str(err)}, status=400)
```

Reply:
That is server code. The client and server are not tightly coupled. You could be using a newer/older version of server with this client code. We should make sure the code doesn't crash when faced with invalid data. 

> "Be conservative in what you send, be liberal in what you accept"
> -- [Robustness Principle](https://en.wikipedia.org/wiki/Robustness_principle)

Reply:
The philosophy is great.

Currently, the server and the client are shipped together. There are changes in the client only when relevant changes are there on the server. This PR is an example of that. When 400 Error was introduced in the server, this needed the creation of 400 handling in the client.

When we break them into different projects, the above may apply as people may use different 
versions of server/client. But since right now server-client is shipped together as a package, we know how the things are going to work out.

Reply:
> Currently, the server and the client are shipped together.

Not really. They are just released together as they are part of the same package. When they are used, the platform may be using a version of firefly server and the user may be using a completely different version of client.